### PR TITLE
build fd from the source

### PR DESCRIPTION
**Pull request by saltedcoffii**
I don't know why we didn't do this first. also fix the license line. also add zsh completion support and fix bash completion support. retain manpage. works on x86_64. needs binaries.

### CODE REVIEW

File: packages/fd.rb [hunk 1]
```diff
@@ -1,42 +1,33 @@
...
+    FileUtils.install "target/release/fd", "#{CREW_DEST_PREFIX}/bin/fd", mode: 0o755
+    FileUtils.install "target/release/build/fd-find-88bf4d216dc4818c/out/fd.bash", "#{CREW_DEST_PREFIX}/share/bash-completion/completions/fd"
+    FileUtils.install "doc/fd.1", "#{CREW_DEST_MAN_PREFIX}/man1/fd.1"
+    FileUtils.install "contrib/completion/_fd", "#{CREW_DEST_PREFIX}/share/zsh/site-functions/_fd"
   end
 
-  def self.postinstall
-    puts "\nTo enable bash completion, execute the following:".lightblue
-    puts "source ~/.bashrc".lightblue
-    puts "\nType 'fd -h' to get started.\n".lightblue
```
Comment (uberhacker) [inline packages/fd.rb:40]: You could keep this last line to help users get started.

Comment (saltedcoffii) [reply packages/fd.rb:40]: it feels unnecessary. the binary is the same name as the package, and --help is the standard help command.

Comment (uberhacker) [reply packages/fd.rb:40]: I don't think it hurts and it probably is a bit pretentious to assume the user knows how to execute.

Comment (saltedcoffii) [reply packages/fd.rb:40]: I disagree. we provide manpages and the `crew files` commands for this kind of lookup information. plus, fd isn't geared toward desktop (likely less experienced) users anyway. @satmandu third opinion?

Comment (satmandu) [reply packages/fd.rb:40]: I'm with @saltedcoffii on this one. I think that telling people to use `fd -h` is ob

Comment (satmandu) [reply packages/fd.rb:40]: I'm with @saltedcoffii on this one. We have already added a man page, so we don't need to suggest people use `fd -h`.

Comment (saltedcoffii) [reply packages/fd.rb:40]: Awesome, let's get some binaries built 🎉

**Comment by satmandu** [comment]
Doesn't build:
```
   Compiling jemallocator v0.3.2
    Finished release [optimized] target(s) in 8m 31s
/usr/local/lib64/ruby/3.0.0/fileutils.rb:880:in `stat': No such file or directory @ rb_file_s_stat - target/release/build/fd-find-88bf4d216dc4818c/out/fd.bash (Errno::ENOENT)
        from /usr/local/lib64/ruby/3.0.0/fileutils.rb:880:in `block in install'
        from /usr/local/lib64/ruby/3.0.0/fileutils.rb:1597:in `block in fu_each_src_dest'
        from /usr/local/lib64/ruby/3.0.0/fileutils.rb:1613:in `fu_each_src_dest0'
        from /usr/local/lib64/ruby/3.0.0/fileutils.rb:1595:in `fu_each_src_dest'
        from /usr/local/lib64/ruby/3.0.0/fileutils.rb:879:in `install'
        from /usr/local/lib/crew/packages/fd.rb:25:in `install'
        from /usr/local/bin/crew:888:in `block in build_and_preconfigure'
        from /usr/local/bin/crew:869:in `chdir'
        from /usr/local/bin/crew:869:in `build_and_preconfigure'
        from /usr/local/bin/crew:1406:in `build_package'
        from /usr/local/bin/crew:1383:in `resolve_dependencies_and_build'
        from /usr/local/bin/crew:1541:in `block in build_command'
        from /usr/local/bin/crew:1537:in `each'
        from /usr/local/bin/crew:1537:in `build_command'
        from /usr/local/bin/crew:1687:in `<main>'
Preconfiguring package...
Command exited with non-zero status 1
```

**Comment by uberhacker** [comment]
What's wrong with the repo provided binaries if they work?  It saves us a lot of work.

**Comment by saltedcoffii** [comment]
@satmandu what architecture did it not build on?

**Comment by satmandu** [comment]
@saltedcoffii all of them.

**Comment by saltedcoffii** [comment]
really? it worked for me. I'll do some testing.

**Comment by saltedcoffii** [comment]
@satmandu this PR should be fixed and still needs binaries

**Comment by satmandu** [comment]
Won't have armv7l builder free until 2022. This `qemu` compile is slow.

@uberhacker maybe you can get to this?

**Comment by uberhacker** [comment]
Tests fail on armv7l.  See below:
```
test test_max_results ... ok
test test_no_ignore_aliases ... ok
test test_no_ignore_vcs ... ok
test test_multi_file ... ok
test test_no_ignore_parent ... ok
test test_normalized_absolute_path ... ok
test test_no_ignore_parent_inner_git ... ok
test test_opposing::absolute_path ... FAILED
test test_opposing::hidden ... ok
test test_opposing::follow ... FAILED
test test_opposing::no_ignore_vcs ... ok
test test_multi_file_with_missing ... ok
test test_opposing::no_ignore ... ok
test test_opposing::u ... FAILED
test test_print0 ... ok
test test_regex_overrides_glob ... ok
test test_number_parsing_errors ... ok
test test_opposing::uu ... FAILED
test test_prune ... ok
test test_regex_searches ... ok
test test_strip_cwd_prefix ... ok
test test_symlink_and_absolute_path ... ok
test test_symlink_and_full_path_abs_path ... ok
test test_symlink_and_full_path ... ok
test test_smart_case_glob_searches ... ok
test test_quiet ... ok
test test_symlink_as_absolute_root ... ok
test test_symlink_as_root ... ok
test test_simple ... ok
test test_type_executable ... ok
test test_smart_case ... ok
test test_type_empty ... ok
test test_type ... ok
test test_size ... ok

failures:

---- test_opposing::absolute_path stdout ----
thread 'test_opposing::absolute_path' panicked at 'assertion failed: `(left == right)`
  left: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./a.foo\n./e1 e2\n./one\n./one/b.foo\n./one/two\n./one/two/C.Foo2\n./one/two/c.foo\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n./symlink\n", stderr: "" }`,
 right: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./symlink\n./a.foo\n./one\n./one/b.foo\n./one/two\n./one/two/c.foo\n./one/two/C.Foo2\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n./e1 e2\n", stderr: "" }`: --relative-path should override --absolute-path', tests/tests.rs:1960:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_opposing::follow stdout ----
thread 'test_opposing::follow' panicked at 'assertion failed: `(left == right)`
  left: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./a.foo\n./e1 e2\n./one\n./one/b.foo\n./one/two\n./one/two/C.Foo2\n./one/two/c.foo\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n./symlink\n", stderr: "" }`,
 right: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./symlink\n./one\n./one/b.foo\n./one/two\n./one/two/c.foo\n./one/two/C.Foo2\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n./a.foo\n./e1 e2\n", stderr: "" }`: --no-follow should override --follow', tests/tests.rs:1960:5

---- test_opposing::u stdout ----
thread 'test_opposing::u' panicked at 'assertion failed: `(left == right)`
  left: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./symlink\n./e1 e2\n./one\n./a.foo\n./one/b.foo\n./one/two\n./one/two/c.foo\n./one/two/C.Foo2\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n", stderr: "" }`,
 right: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./symlink\n./a.foo\n./e1 e2\n./one\n./one/b.foo\n./one/two\n./one/two/c.foo\n./one/two/C.Foo2\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n", stderr: "" }`: --ignore should override -u', tests/tests.rs:1960:5

---- test_opposing::uu stdout ----
thread 'test_opposing::uu' panicked at 'assertion failed: `(left == right)`
  left: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./a.foo\n./symlink\n./e1 e2\n./one\n./one/b.foo\n./one/two\n./one/two/c.foo\n./one/two/C.Foo2\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n", stderr: "" }`,
 right: `Output { status: ExitStatus(unix_wait_status(0)), stdout: "./a.foo\n./e1 e2\n./one\n./one/b.foo\n./one/two\n./one/two/C.Foo2\n./one/two/c.foo\n./one/two/three\n./one/two/three/d.foo\n./one/two/three/directory_foo\n./symlink\n", stderr: "" }`: --ignore --no-hidden should override -uu', tests/tests.rs:1960:5


failures:
    test_opposing::absolute_path
    test_opposing::follow
    test_opposing::u
    test_opposing::uu

test result: FAILED. 68 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out; finished in 12.62s

error: test failed, to rerun pass '--test tests'
/usr/local/lib/crew/lib/package.rb:180:in `system': Command failed with exit 101: bash -e -c "cargo test --release"
fd failed to build: `cargo test --release` exited with 101
```

**Comment by uberhacker** [comment]
Should we go back to the binary releases?

**Comment by saltedcoffii** [comment]
can you post the error logs for those failed tests? I'm sure there's an explanation.

**Comment by satmandu** [comment]
@uberhacker that failure is due to the old `package.rb`.

@saltedcoffii  can you merge master?

**Comment by satmandu** [comment]
armv7l tests pass for me:
```
test result: ok. 72 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.09s
```

**Pull request merged**